src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/ManifestVersionUpdate.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.NET.Sdk.WorkloadManifestReader
+{
+    public class ManifestVersionUpdate : IEquatable<ManifestVersionUpdate>, IComparable<ManifestVersionUpdate>
+    {
+        public ManifestVersionUpdate(ManifestId manifestId, ManifestVersion? existingVersion, string? existingFeatureBand, ManifestVersion? newVersion, string? newFeatureBand)
+        {
+            ManifestId = manifestId;
+            ExistingVersion = existingVersion;
+            ExistingFeatureBand = existingFeatureBand;
+            NewVersion = newVersion;
+            NewFeatureBand = newFeatureBand;
+        }
+
+        public ManifestId ManifestId { get; }
+        public ManifestVersion? ExistingVersion { get; }
+        public string? ExistingFeatureBand { get; }
+        public ManifestVersion? NewVersion { get; }
+        public string? NewFeatureBand { get; }
+
+        //  Returns an object representing an undo of this manifest update
+        public ManifestVersionUpdate Reverse()
+        {
+            return new ManifestVersionUpdate(ManifestId, NewVersion, NewFeatureBand, ExistingVersion, ExistingFeatureBand);
+        }
+
+        public int CompareTo(ManifestVersionUpdate? other)
+        {
+            if (other == null) return 1;
+            int ret = ManifestId.CompareTo(other.ManifestId);
```

Comment:
For a version update of a manifest, shouldn't the IDs always be the same and the underlying versions be different. Should we instead throw something like `InvalidOperationException` if the IDs are different?

Reply:
This is just implemented so that equality works correctly, so that in tests you can assert that the manifest updates were what was expected.